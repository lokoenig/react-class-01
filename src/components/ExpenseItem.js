import React from "react";
import { Link } from 'react-router-dom';



import './ExpenseItem.scss';
import EditExpenseButton from "./EditExpenseButton";

// Description, amount, creation date
const ExpenseItem = (props) => (
    <Link
        to={'/edit/' + props.id}
        className="expense-single"
        key={props.id}
    >
        <span className="expense-single__description">{props.description}</span>
        <span className="expense-singe__created">
            {props.created.toLocaleDateString()}
        </span>
        <span className="expense-single__amount">
            {new Intl.NumberFormat(
                'en-US',
                { style: 'currency', currency: 'USD' }
            ).format(props.amount / 100)}

        </span>

    </Link >
);

export default ExpenseItem;