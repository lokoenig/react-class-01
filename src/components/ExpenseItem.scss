.expense-single {
    display: grid;
    width: 30rem;
    grid-template-columns: 10rem 1fr 1fr max-content;
    gap: 1rem;
    &>span {
        outline: 1px solid plum;
    }
}