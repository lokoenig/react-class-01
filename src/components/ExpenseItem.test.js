import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";

import ExpenseSingleTestData from "../fixtures/expense-single";

import ExpenseItem from "./ExpenseItem";


test('ExpenseItem - render single item', ()=>{
    const { getByText, asFragment } = 
        render(
            <BrowserRouter>
            <ExpenseItem 
                id={ExpenseSingleTestData.id}
                amount={ExpenseSingleTestData.amount}
                created={ExpenseSingleTestData.created}
                note={ExpenseSingleTestData.note}
                description={ExpenseSingleTestData.description}
            />
            </BrowserRouter>
        );
    
  //  screen.debug();
    expect(asFragment()).toMatchSnapshot();
});