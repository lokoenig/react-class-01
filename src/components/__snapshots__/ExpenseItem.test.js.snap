// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseItem - render single item 1`] = `
<DocumentFragment>
  <div
    class="expense-single"
  >
    <span
      class="expense-single__description"
    >
      A very special expense
    </span>
    <span
      class="expense-single__amount"
    >
      $666.66
    </span>
    <span
      class="expense-singe__created"
    >
      3/9/2023
    </span>
    <button>
      Edit
    </button>
  </div>
</DocumentFragment>
`;
